test/jdk/sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA.java

```diff
@@ -0,0 +1,185 @@
+            + "AAADQAsACoEAwUDBgMIBwgICAQIBQgGCAkICggLBAEFAQYBBAIDAwMBAwICAwIBA"
+            + "gIAKwAFBAMEAwMALQACAQEAMgAsACoEAwUDBgMIBwgICAQIBQgGCAkICggLBAEFA"
+            + "QYBBAIDAwMBAwICAwIBAgIALwBrAGkAHQAAAAARACAAZMUAADkwsiaOwcsWAwAAA"
+            + "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+            + "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+            + "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+            + "AAAAAAtAAAAAAAAAAEAADAAAAA=");
+
+    /*
+     * The following is to set up the keystores.
+     */
+    private static final String passwd = "passphrase";
+
+    /*
+     * Main entry point for this demo.
+     */
+    public static void main(String[] args) throws Exception {
+        if (debug) {
+            System.setProperty("javax.net.debug", "all");
+        }
+
+        TestBadDNForPeerCA test = new TestBadDNForPeerCA();
+
+        try {
+            test.runTest(payload);
+            throw new Exception(
+                "TEST FAILED:  Didn't generate any exception");
+        } catch (SSLHandshakeException she) {
+            System.out.println("TEST PASSED:  Caught expected exception");
+        }
+    }
+
+    /*
+     * Create an initialized SSLContext to use for this demo.
+     */
+
+    public TestBadDNForPeerCA() throws Exception {
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+        KeyStore ts = KeyStore.getInstance("JKS");
+
+        char[] passphrase = passwd.toCharArray();
+
+        ks.load(new FileInputStream(keyStoreFile), passphrase);
+        ts.load(new FileInputStream(keyStoreFile), passphrase);
+
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
+        kmf.init(ks, passphrase);
+
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+        tmf.init(ts);
+
+        SSLContext sslCtx = SSLContext.getInstance(proto);
+
+        sslCtx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+
+        sslc = sslCtx;
+    }
+
+
+    private void runTest(byte[] injected) throws Exception {
+
+        createSSLEngines();
+        createBuffers();
+
+        cTOs = ByteBuffer.wrap(injected);
+
+        System.out.println("injecting client hello");
+
+        for (int i = 0; i < 10; i++) { //retry if survived
+            SSLEngineResult serverResult = serverEngine.unwrap(cTOs, serverIn);
+            System.out.println("server unwrap: " + serverResult);
+            runDelegatedTasks(serverResult, serverEngine);
+        }
+    }
+
+    private void createSSLEngines() throws Exception {
+
+        serverEngine = sslc.createSSLEngine();
+        serverEngine.setUseClientMode(false);
+        serverEngine.setNeedClientAuth(true);
+
+    }
+
+
+    private void createBuffers() {
+
+        serverIn = ByteBuffer.allocateDirect(65536);
+
+        cTOs = ByteBuffer.allocateDirect(65536);
```

Comment:
not used - you immediately overwrite this value in runTest

Reply:
He updated the Template code but didn't take all of the unused stuff out.  I'm ok with it either way.